daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -1503,7 +1718,92 @@ func (devices *DeviceSet) DeleteDevice(hash string) error {
 	devices.Lock()
 	defer devices.Unlock()
 
-	return devices.deleteDevice(info)
+	// If mountcount is not zero, that means devices is still in use
+	// or has not been Put() properly. Fail device deletion.
+
+	if info.mountCount != 0 {
+		return fmt.Errorf("devmapper: Can't delete device %v as it is still mounted. mntCount=%v", info.Hash, info.mountCount)
```

Comment:
is this going to be different functionality than currently?
Or is this just returning earlier from what would be an EBUSY?